oshi-core/src/main/java/oshi/hardware/platform/unix/solaris/SolarisComputerSystem.java

```diff
@@ -119,70 +141,48 @@ private static SmbiosStrings readSmbios() {
-            if (checkLine.contains("SMB_TYPE_") && (smbTypeId = getSmbType(checkLine)) == Integer.MAX_VALUE) {
+            if (checkLine.contains("SMB_TYPE_") && (smbTypeId = getSmbType(checkLine)) == null) {
                 // If we get past what we need, stop iterating
                 break;
             }
             // Based on the smbTypeID we are processing for
-            switch (smbTypeId) {
-            case 0: // BIOS
-                if (checkLine.contains(vendorMarker)) {
-                    biosVendor = checkLine.split(vendorMarker)[1].trim();
-                } else if (checkLine.contains(biosVersionMarker)) {
-                    biosVersion = checkLine.split(biosVersionMarker)[1].trim();
-                } else if (checkLine.contains(biosDateMarker)) {
-                    biosDate = checkLine.split(biosDateMarker)[1].trim();
-                }
-                break;
-            case 1: // SYSTEM
-                if (checkLine.contains(manufacturerMarker)) {
-                    manufacturer = checkLine.split(manufacturerMarker)[1].trim();
-                } else if (checkLine.contains(productMarker)) {
-                    model = checkLine.split(productMarker)[1].trim();
-                } else if (checkLine.contains(serialNumMarker)) {
-                    serialNumber = checkLine.split(serialNumMarker)[1].trim();
-                } else if (checkLine.contains(uuidMarker)) {
-                    uuid = checkLine.split(uuidMarker)[1].trim();
-                }
-                break;
-            case 2: // BASEBOARD
-                if (checkLine.contains(manufacturerMarker)) {
-                    boardManufacturer = checkLine.split(manufacturerMarker)[1].trim();
-                } else if (checkLine.contains(productMarker)) {
-                    boardModel = checkLine.split(productMarker)[1].trim();
-                } else if (checkLine.contains(versionMarker)) {
-                    boardVersion = checkLine.split(versionMarker)[1].trim();
-                } else if (checkLine.contains(serialNumMarker)) {
-                    boardSerialNumber = checkLine.split(serialNumMarker)[1].trim();
-                }
-                break;
-            default:
-                break;
+            if (smbTypeId != null) {
+                String key = checkLine.substring(0, checkLine.indexOf(":")).trim();
+                String val = checkLine.substring(checkLine.indexOf(":") + 1).trim();
+                smbTypesMap.get(smbTypeId).put(key, val);
             }
         }
+
         // If we get to end and haven't assigned, use fallback
-        if (Util.isBlank(serialNumber)) {
-            serialNumber = readSerialNumber();
+        if (!smbTypesMap.get(SmbType.SMB_TYPE_SYSTEM).containsKey(serialNumMarker)
+                || Util.isBlank(smbTypesMap.get(SmbType.SMB_TYPE_SYSTEM).get(serialNumMarker))) {
+            smbTypesMap.get(SmbType.SMB_TYPE_SYSTEM).put(serialNumMarker, readSerialNumber());
```

Comment:
Can we do this `get(SMB_TYPE_SYSTEM)` just once to avoid this repetition?  Also why not make those enums static imports?